graphrag/index/api.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (c) 2024 Microsoft Corporation.
+# Licensed under the MIT License
+
+"""
+Indexing API for GraphRAG.
+
+WARNING: This API is under development and may undergo changes in future releases.
+Backwards compatibility is not guaranteed at this time.
+"""
+
+import logging
+from pathlib import Path
+
+from graphrag.config.enums import CacheType, ReportingType
+from graphrag.config.models.graph_rag_config import GraphRagConfig
+from graphrag.config.resolve_timestamp_path import resolve_timestamp_path
+
+from .cache.noop_pipeline_cache import NoopPipelineCache
+from .create_pipeline_config import create_pipeline_config
+from .emit.types import TableEmitterType
+from .progress import (
+    NullProgressReporter,
+    PrintProgressReporter,
+    ProgressReporter,
+)
+from .progress.rich import RichProgressReporter
+from .run import run_pipeline_with_config
+from .typing import PipelineRunResult
+
+
+def enable_logging(log_filepath: str | Path, verbose: bool = False) -> None:
+    """Enable logging to a file.
+
+    Parameters
+    ----------
+    log_filepath : str | Path
+        The path to the log file.
+    verbose : bool, default=False
+        Whether to log debug messages.
+    """
+    log_filepath = Path(log_filepath)
+    log_filepath.parent.mkdir(parents=True, exist_ok=True)
+    log_filepath.touch(exist_ok=True)
+
+    logging.basicConfig(
+        filename=log_filepath,
+        filemode="a",
+        format="%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s",
+        datefmt="%H:%M:%S",
+        level=logging.DEBUG if verbose else logging.INFO,
+    )
+
+
+def enable_logging_with_config(
+    config: GraphRagConfig, timestamp_value: str, verbose: bool = False
+) -> tuple[bool, str]:
+    """Enable logging to a file based on the config.
+
+    Parameters
+    ----------
+    config : GraphRagConfig
+        The configuration.
+    timestamp_value : str
+        The timestamp value representing the directory to place the log files.
+    verbose : bool, default=False
+        Whether to log debug messages.
+
+    Returns
+    -------
+    tuple[bool, str]
+        A tuple of a boolean indicating if logging was enabled and the path to the log file.
+        (False, "") if logging was not enabled.
+        (True, str) if logging was enabled.
+    """
+    if config.reporting.type == ReportingType.file:
+        log_path = resolve_timestamp_path(
+            Path(config.root_dir) / config.reporting.base_dir / "indexing-engine.log",
+            timestamp_value,
+        )
+        enable_logging(log_path, verbose)
+        return (True, str(log_path))
+    return (False, "")
+
+
+def load_progress_reporter(reporter_type: str = "none") -> ProgressReporter:
```

Comment:
Can this function be moved to `cli.py` as a private function`_load_progress_reporter(...)`? It doesn't look like the index API ever relies on calling this function...only the CLI does.

Reply:
Disregard this comment. I see now that a user would need to call this function first in order to pass the right object into `build_index(...)`.

I think the other logging functions though are only useful to the CLI though and shouldn't be part of the API right?

Reply:
Moved the load progress code to the progress module.